### ISSUE TITLE
DW1560 Bluetooth won't connect to devices

### ISSUE THREAD

**Issue by d3myz** [bavariancake/XPS9570-macOS#25]
Hi All, I tried connecting a couple different BT headphones to the Hack and it tells me it can't connect because of the PIN. PIN is set to 0000 and it will find the device, not connect and then won't find any other devices, i've turned on/off debugged, reset etc. I noticed when looking at the guide that I don't have OS-X-BcrmPatchRam in my kexts and neither BrcmPatchRAM2.kext or BrcmFirmwareRepo.kext are installed in the L/E directory, but are in Clover/Kexts/Other Airdrop doesn't work either, but WiFi is fine. I downloaded OS-X-BcrmPatchRam but it doesn't look like a kext, do I have to compile it? I apologize for being a n00b. I've never compiled a kext before. Anyways, Big Thanks to @bavariancake for this great build and the community. $
<img width="255" alt="Screen Shot 2019-06-06 at 3 00 39 PM" src="https://user-images.githubusercontent.com/50464405/59066285-e799b800-886b-11e9-82c5-863066199148.png">

**Comment by d3myz** [comment]
Well, I just solved my own problem. I understand now that there are 2 kexts to the OS-XBcrmPatchRam, and I registered them in L/E and now the bluetooth works like a charm.

**Comment by prajals** [comment]
> Well, I just solved my own problem. I understand now that there are 2 kexts to the OS-XBcrmPatchRam, and I registered them in L/E and now the bluetooth works like a charm. Another thing i noticed that by doing this the kext updated the firmware for my BT. it was 4096 now it's 5747.
> <img alt="Screen Shot 2019-06-06 at 4 32 19 PM" width="249" src="https://user-images.githubusercontent.com/50464405/59070652-ca1f1b00-8878-11e9-9350-db88e4ea0c71.png">

hello @d3myz  have you checked with your bluetooth headset or speakers? i am having crackling audio white using bluetooth spears or headsets.
Thanks

**Comment by d3myz** [comment]
Yes, I have been having intermittent sound quality issues with my headsets while using facetime and also loss of audio while disconnecting them. I'm not sure what the issue might be, but I have read that the dw1830 is a better wifi card all around especially for dual boot setups. the 1830 also support bt 4.1 which should have better quality.
